characteristic_secure/device_code.js

```diff
@@ -0,0 +1,144 @@
+  const msg = 'Dbg: ' + debug;
+  g.drawString(msg, g.getWidth() - g.stringWidth(msg), g.getHeight() - 10);
+  g.setFontVector(40);
+  const val = `${ageData[0]}:${heartRateMaxData[0]}`;
+  g.drawString(val, (g.getWidth() - g.stringWidth(val)) / 2, 12);
+
+  g.flip();
+}
+
+function onInit() {
+  console.log(`Starting onInit. passkey: "${devicePassKey}"`);
+  NRF.setServices({
+    '00002a80-0000-1000-8000-00805f9b34fb': {   // "age" service.
+      '00002a80-0000-1000-8000-00805f9b34fb': { // "age" characteristic.
+        value: ageData.buffer,
+        broadcast: false,
+        readable: true,
+        writable: true,
+        notify: false,
+        description: 'Age',
+        onRead: function () {
+          debug = 'onRead';
+          setTimeout(updateScreen(), 100);
+          return ageData.buffer;
+        },
+        onWrite: function (evt) {
+          ageData[0] = new DataView(evt.data).getUint8(0, /*littleEndian=*/true);
+          debug = 'onWrite';
+          setTimeout(updateScreen(), 100);
+        },
+        security: {
+          read: {
+            encrypted: true,  // Encrypt data (default: false).
+            mitm: true,       // Man In The Middle (default: false).
+          },
+          write: {
+            encrypted: true,  // Encrypt data (default: false).
+            mitm: true,       // Man In The Middle (default: false).
+          }
+        }
+      }
+    },
+    '0000180d-0000-1000-8000-00805f9b34fb': {   // "heart_rate" service.
+      '00002a8d-0000-1000-8000-00805f9b34fb': { // "heart_rate_max" characteristic.
+        value: heartRateMaxData.buffer,
+        broadcast: false,
+        readable: true,
+        writable: true,
+        notify: false,
+        description: 'Maximum heart rate',
+        onRead: function () {
+          debug = 'onRead';
+          setTimeout(updateScreen(), 100);
+          return heartRateMaxData.buffer;
+        },
+        onWrite: function (evt) {
+          heartRateMaxData[0] = new DataView(evt.data).getUint8(0, /*littleEndian=*/true);
+          debug = 'onWrite';
+          setTimeout(updateScreen(), 100);
+        },
+        security: {
+          read: {
+            encrypted: false,  // Encrypt data (default: false).
+            mitm: false,       // Man In The Middle (default: false).
+          },
+          write: {
+            encrypted: false,  // Encrypt data (default: false).
+            mitm: false,       // Man In The Middle (default: false).
+          }
+        }
+      }
+    }
+  });
+
+  NRF.on('disconnect', (reason) => {
+    debug = `disconnect: ${reason}`;
+    isConnected = false;
+    updateUI();
+  });
+
+  NRF.on('connect', (addr) => {
+    debug = `onconnect: ${addr}`;
+    isConnected = true;
+    updateUI();
+  });
+
+  NRF.setSecurity({
+    display: deviceHasDisplay(),  // Can this device display a passkey?
+    keyboard: false,              // Can this device enter a passkey?
```

Comment:
Use `deviceHasKeyboard()`.

Reply:
I deleted the function as it didn't improve clarity.